src/safeds/data/tabular/containers/_table.py

```diff
@@ -1069,6 +1069,8 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
                         key={val: ix for ix, val in enumerate(self.column_names)}.__getitem__,
                     ),
                 )
+            if self.number_of_rows == 0:
```

Comment:
```suggestion
            if rows.number_of_rows == 0:
                return self
            if self.number_of_rows == 0:
```

Then we also don't concat if there's nothing to do, the other way round